paprika_recipes/remote.py

```diff
@@ -75,6 +76,30 @@ def get_recipe_by_id(self, id: str, hash: str) -> RemoteRecipe:
     def count(self) -> int:
         return len(self._get_remote_recipe_identifiers())
 
+    def download_image(self, recipe: RemoteRecipe):
+
+        if recipe.image_url:
+            result = requests.request('GET', recipe.image_url)
+            result.raise_for_status()
+
+            if not os.path.exists('images'):
+                os.mkdir('images')
+
+            url = recipe.image_url
+
+            # Remove query parameters from the URL.
+            if url.find('?') != -1:
+                url = url[:url.find('?')]
+
+            filename, file_extension = os.path.splitext(url)
+            destination_filename = f"images/{recipe.uid}{file_extension}"
+
+            file = open(destination_filename, "wb")
+            file.write(result.content)
+            file.close()
+
+            recipe.image_url = destination_filename
```

Comment:
I think if you change this, when you try to re-upload the recipe, the recipe will be pointing at your local file path instead of the correct URL, no?

Reply:
@coddingtonbear Given that I went with a new property for now. I'm currently looking what paprika is actually doing when uploading images though